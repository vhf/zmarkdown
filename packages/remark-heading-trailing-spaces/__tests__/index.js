import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import rehypeStringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

import plugin from '../src/'


const render = text => unified()
  .use(reParse)
  .use(remark2rehype)
  .use(plugin)
  .use(rehypeStringify)
  .processSync(text.replace(/·/g, ' '))

/**
 * (for convenience, · are replaced with
 * simple single spaces in the tests)
 */

const fixture = dedent`
  Header
  ------··

  ### H3···

  H1··
  =

  H2
  --··
`

test('with plugin', () => {
  const { contents } = render(fixture)
  expect(contents).toMatchSnapshot()
})

test('without', () => {
  const { contents } = unified()
    .use(reParse)
    .use(remark2rehype)
    .use(rehypeStringify)
    .processSync(fixture)

  expect(contents).toMatchSnapshot()
})

test('regression 1', () => {
  const { contents } = render(dedent`
    word
    -·item·1
    -·item·2

    a|b
    ---|---
    c|d
  `)
  expect(contents).toMatchSnapshot()
})
