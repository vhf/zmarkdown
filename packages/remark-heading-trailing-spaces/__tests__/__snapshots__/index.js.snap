// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regression 1 1`] = `
"<p>word</p>
<ul>
<li>item 1</li>
<li>item 2</li>
</ul>
<table>
<thead>
<tr>
<th>a</th>
<th>b</th>
</tr>
</thead>
<tbody>
<tr>
<td>c</td>
<td>d</td>
</tr>
</tbody>
</table>"
`;

exports[`with plugin 1`] = `
"<h2>Header</h2>
<h3>H3</h3>
<h1>H1</h1>
<p>H2
--  </p>"
`;

exports[`without 1`] = `
"<p>Header
------··</p>
<h3>H3···</h3>
<h1>H1··</h1>
<p>H2
--··</p>"
`;
